### DISCUSSION TITLE
Local plugins development

### DISCUSSION THREAD

**Discussion by KonradLinkowski** [parcel-bundler/parcel#6939] [Q&A]
I wanted to make a simple plugin for parcel v2, another layer of minification basically.
The first problem is that I'm not really sure if it should be an [optimizer ](https://v2.parceljs.org/plugin-system/optimizer/) or a [compressor](https://v2.parceljs.org/plugin-system/compressor/) seems the same for me.

The second one is that I don't know how to test a plugin locally. Should I have two separate folders one with the plugin code and the second one with an example code? Then I could run `npm link` on the plugin and use it in that second folder, but this doesn't seem a reliable solution.

Any suggestion or example plugin with a well-configured environment?

**Comment by mischnic** [comment]
If you want to do things like precompressing files with gzip for your server, use a compressor.
If you want to do minification like Terser or any other kind of per-bundle transformation, use an optimizer.

See https://v2.parceljs.org/configuration/plugin-configuration/#local-plugins

**Reply by KonradLinkowski** [reply]
Thank you for the clarification 😄
So this link describes exactly what I was thinking about. I thought that maybe there is some more organized solution so multiple people can work on a plugin. But this makes sense as well.

**Reply by KonradLinkowski** [reply]
Seems like compressors exist in the documentation and on the *v2* branch, but there isn't an npm version with it included yet.

**Reply by mischnic** [reply]
You probably need to use the nightly version (`yarn add parcel@nightly`)
